src/node/commit.rs

```diff
@@ -538,4 +538,18 @@ mod tests {
         // I created this unit test by hand
         assert_program_not_deserializable::<Core>(&[0xa9, 0x48, 0x00], &Error::NotInCanonicalOrder);
     }
+
+    #[test]
+    fn regression_177_1() {
+        // `case (drop iden) iden` from upstream occurs-check test. Has an infinitely sized
+        // input type. Will fail trying to unify the input type with the unit type *before*
+        // doing the occurs check, putting an infinitely-sized type into the error variant.
+        // Check that this error type can be printed in finite space/time.
+        let mut iter = BitIter::from([0xc1, 0x07, 0x20, 0x30].iter().copied());
```

Comment:
50293ca9872e6ad6dddd42aa020f17a7d43336e6 This test case would be better maintainable if we wrote it in the human encoding. We should strive to remove all byte blobs from the code base.

Reply:
I'll take a look, but our options for decoding `ConstructNode`s are pretty limited.